R/plumber.R

```diff
@@ -767,13 +767,31 @@ Plumber <- R6Class(
         function(...) {
           resetForward()
           # TODO: support globbing?
-
           if (nchar(path) >= nchar(mountPath) && substr(path, 0, nchar(mountPath)) == mountPath) {
-            # This is a prefix match or exact match. Let this router handle.
+            # This is a prefix match or exact match. Let mount attempt handle.
+
+            # Mark that the route is being handled within a mount.
+            # Allows for the mount to forward to the next mount instead of 404.
+            prev_mount_status <- req$`_IS_MOUNT`
+            req$`_IS_MOUNT` <- TRUE
 
             # First trim the prefix off of the PATH_INFO element
+            cur_path_info <- req$PATH_INFO
             req$PATH_INFO <- substr(req$PATH_INFO, nchar(mountPath), nchar(req$PATH_INFO))
-            return(private$mnts[[mountPath]]$route(req, res))
+
+            # Handle route
+            ret <- private$mnts[[mountPath]]$route(req, res)
+
+            # Undo path info and mount status changes
+            req$PATH_INFO <- cur_path_info
+            req$`_IS_MOUNT` <- prev_mount_status
```

Comment:
req$PATH_INFO reset needs to happen in a separate step (so that async is supported)